server/models/Expense.js

```diff
@@ -365,6 +365,11 @@ function defineModel() {
   Expense.prototype.setPaid = async function (lastEditedById) {
     await this.update({ status: status.PAID, lastEditedById });
 
+    // Update transactions settlement
+    if (this.data?.['isPlatformTipSettlement']) {
```

Comment:
Note for the future, this value `isPlatformTipSettlement` doesn't look good now.

We should move to something cleaner and explicit like Expense.type = 'SETTLEMENT'

Reply:
I agree